docs/source/notes/randomness.rst

```diff
@@ -2,70 +2,73 @@ Reproducibility
+sure to give them a fixed seed.
 
-PyTorch
-.......
+Random number generation
+........................
 You can use :meth:`torch.manual_seed()` to seed the RNG for all devices (both
 CPU and CUDA)::
 
     import torch
     torch.manual_seed(0)
 
+Deterministic algorithms
+........................
+Some PyTorch operations use nondeterministic algorithms by default. A subset
+of them can be configured to use a deterministic implementation or throw an
+error if a deterministic implementation is not available. Please check the
+documentation for :meth:`torch.set_deterministic()` for a full list of affected
+operations. If an operation does not act correctly according to the documentation,
+or if you need a deterministic implementation of an operation that does not have
+one, please submit an issue:
+`<https://github.com/pytorch/pytorch/issues?q=label:%22topic:%20determinism%22>`_
 
-There are some PyTorch functions that use CUDA functions that can be a source
-of nondeterminism. One class of such CUDA functions are atomic operations,
-in particular :attr:`atomicAdd`, which can lead to the order of additions being
-nondetermnistic. Because floating-point addition is not perfectly associative
-for floating-point operands, :attr:`atomicAdd` with floating-point operands can
-introduce different floating-point rounding errors on each evaluation, which
-introduces a source of nondeterministic variance (aka noise) in the result.
+.. code-block::
 
-PyTorch functions that use :attr:`atomicAdd` in the forward kernels include
-:meth:`torch.Tensor.index_add_`, :meth:`torch.Tensor.scatter_add_`,
-:meth:`torch.bincount`.
+    torch.set_deterministic(True)
 
-A number of operations have backwards kernels that use :attr:`atomicAdd`,
-including :meth:`torch.nn.functional.embedding_bag`,
-:meth:`torch.nn.functional.ctc_loss`, :meth:`torch.nn.functional.interpolate`,
-and many forms of pooling, padding, and sampling.
+CUDA convolution
+----------------
+When running convolution operations CUDA devices, two options must be set::
 
-There is currently no simple way of avoiding nondeterminism in these functions.
-
-Additionally, the backward path for :meth:`repeat_interleave` operates
-nondeterministically on the CUDA backend because :meth:`repeat_interleave`
-is implemented using :meth:`index_select`, the backward path for
-which is implemented using :meth:`index_add_`, which is known to operate
-nondeterministically (in the forward direction) on the CUDA backend (see above).
+    torch.backends.cudnn.deterministic = True
+    torch.backends.cudnn.benchmark = False
 
-cuDNN
-.....
-When running on the cuDNN backend, two further options must be set::
+Alternatively, you can use the following, since CUDA convolution will be
+deterministic if either :code:`torch.set_deterministic(True)` or
+:code:`torch.backends.cudnn.deterministic = True` is set::
 
-    torch.backends.cudnn.deterministic = True
+    torch.set_deterministic(True)
```

Comment:
This is really interesting but also, I think, consistent with the design of set_deterministic. That is, we could imagine that set_deterministic would also set `torch.backends.cudnn.benchmark = False`, and if it did then I would expect `set_deterministic` to ALSO set PyTorch's generator seed:

```
torch.set_deterministic(True, seed=1234)
```

That is, it would be a more "all in one" solution. But `set_deterministic` actually has a narrower scope as described in the previous section.

I wonder if a sentence clarifying what `set_deterministic` is not might be helpful in the previous section?

Reply:
I added a sentence in the previous section that points to this explanation of the benchmark flag.

I like your idea of including a `seed` kwarg in `set_deterministic`. I can add that feature in another PR.

My comment here (https://github.com/pytorch/pytorch/issues/15359#issuecomment-659658715), and ngimel's comment above that explains why I thought it makes sense for `set_deterministic` not to change the benchmark setting. But I don't have any strong opinions about it. It would make things a little less complicated for most users if we do include it.